packages/discord.js/typings/index.d.ts

```diff
@@ -441,17 +444,20 @@ export abstract class Application extends Base {
 export class ApplicationCommand<PermissionsFetchType = {}> extends Base {
   private constructor(client: Client<true>, data: RawApplicationCommandData, guild?: Guild, guildId?: Snowflake);
   public applicationId: Snowflake;
+  public contexts: InteractionContextType[] | null;
   public get createdAt(): Date;
   public get createdTimestamp(): number;
   public defaultMemberPermissions: Readonly<PermissionsBitField> | null;
   public description: string;
   public descriptionLocalizations: LocalizationMap | null;
   public descriptionLocalized: string | null;
+  /** @deprecated Use {@link ApplicationCommand.contexts} instead */
   public dmPermission: boolean | null;
   public guild: Guild | null;
   public guildId: Snowflake | null;
   public get manager(): ApplicationCommandManager;
   public id: Snowflake;
+  public integrationTypes: ApplicationIntegrationType[] | null;
```

Comment:
Should this be read-only? We should not be able to modify an array that represents API data, right? @vladfrangu

Reply:
What do we do in other places? But yes, realistically it should be readonly (but that opens up a can of discussion whether all collections and such should also return readonly states)

Reply:
It seems others are mutable. I'm fine with leaving this as-is then.